### PR TITLE
Add GMS Parser

### PR DESCRIPTION
**Pull request by kdeloach**
This adds a class called `GmsReader` which is able to parse version
1.4.0 GMS files into a dictionary object.

Connects #4
Connects #8

### CODE REVIEW

File: gwlfe/gwlfe.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    HAMMON_METHOD = 0
+    BLAINY_CRIDDLE_METHOD = 1
+
+
+class GROW_FLAG(object):
+    NON_GROWING_SEASON = 0
+    GROWING_SEASON = 1
+
+
+class LANDUSE(object):
```
Comment (mmcfarland) [inline gwlfe/gwlfe.py]: Are there `int` codes that identify these values?  We were provided an NLCD code to "MapShed Land Use" code table:  https://drive.google.com/open?id=0B3v0QxIOuR_nX3Rnekp0NGUyOGM.  Might be best to just note that we have such a lookup table for now.

Comment (kdeloach) [inline gwlfe/gwlfe.py]: These land use values are specified by strings in the GMS file but there will likely be a transformation phase where we change this data to a more suitable format.

**Comment by mmcfarland** [comment]
This looks really great.  I think keeping they variable names from the original will help us port over code and get the right values from this dictionary, good call on that.

**Comment by kdeloach** [comment]
Test by running:

```
$ ./gwlfe.py sample_input.gms
```

**Comment by kdeloach** [comment]
Sample files removed. Test by using the sample GMS file from Google Drive.

**Comment by mmcfarland** [comment]
+1 This parses the sample, plus I did a ton of spot checks between the GMS, the dictionary and the key file document.  All of the tested values matched.

**Comment by caseypt** [comment]
+1. Tested. Looks good to me. Nice comments.

File: gwlfe/gwlfe.py [hunk 2]
```diff
@@ -0,0 +1,974 @@
...
+            return LandUse.HD_RESIDENTIAL
+        raise ValueError('Unexpected value: ' + str(value))
+
+
+def iterate_csv_values(fp):
+    """
+    Yield values as a continuous stream for each line in a
+    file-like object.
+    """
+    fp = csv.reader(fp)
```
Comment (rajadain) [inline gwlfe/gwlfe.py]: This local variable has the same name as the parameter which is a little confusing

**Comment by rajadain** [comment]
Just confirming that lines 53-70 are being translated correctly, since there was a discrepancy in documentation and code for them ...

File: gwlfe/gwlfe.py (hunk 2 above)
Comment (kdeloach) [inline gwlfe/gwlfe.py]: Updated.

**Comment by rajadain** [comment]
Unfortunately there is mistake in the documentation for lines 53 - 70, noted in cell A75 of the [GMS Key](https://docs.google.com/spreadsheets/d/12hsklRHwl6sNum0xPk6d5Cw6ONDA3X6BR5-Pc-NwJBc). While the documentation claims that the rows are grouped by contaminant first, and then by land use

```
Lines 53 - 58: (for each Urban Land Use Category, Nitrogen Contaminant)
Lines 59 - 64: (for each Urban Land Use Category, Phosphorus Contaminant)
Lines 65 - 70: (for each Urban Land Use Category, Sediment Contaminant)
```

this is not the case. Actually, they are grouped by land use first, and then by contaminant:

```vb
' Write N, P, and Sed code lines (used to identify the following N, P, and Sed values)
GwlfFileWriter.WriteLine("Nitrogen")
GwlfFileWriter.WriteLine("Phosphorus")
GwlfFileWriter.WriteLine("Sediment")

' Write N, P and Sed values to file
For l = 11 To 16
    GwlfFileWriter.WriteLine(NLoadRate(l))
    GwlfFileWriter.WriteLine(PLoadRate(l))
    GwlfFileWriter.WriteLine(SedLoadRate(l))
Next
```

In `input_4.gms:50-70` we have:

```
"Nitrogen"
"Phosphorus"
"Sediment"
.095,.015,.33,0
.0095,.0021,.4,0
2.8,.8,0,0
.105,.015,.33,0
.0105,.0021,.4,0
6.2,.8,0,0
.11,.015,.33,0
.0115,.0021,.4,0
2.8,.8,0,0
.095,.015,.28,0
.0095,.0019,.37,0
2.5,1.3,0,0
.1,.015,.28,0
.0115,.0039,.37,0
6.2,1.1,0,0
0,0,0,0
0,0,0,0
0,0,0,0
```

But the output of the parser for `LoadRateImp` is:

```json
"LoadRateImp": [
    [0.095,0.11,0.1],
    [0.0095,0.0115,0.0115],
    [2.8,2.8,6.2],
    [0.105,0.095,0],
    [0.0105,0.0095,0],
    [6.2,2.5,0]
],
```

While it should be:

```json
"LoadRateImp": [
    [0.095,0.0095,2.8],
    [0.105,0.0105,6.2],
    [0.11,0.0115,2.8],
    [0.095,0.0095,2.5],
    [0.1,0.015,0.28],
    [0,0,0]
],
```

**Comment by kdeloach** [comment]
That's confusing but it should be fixed now.

**Comment by rajadain** [comment]
Taking another look ...

**Comment by kdeloach** [comment]
Squashed.

**Comment by rajadain** [comment]
+1 looks great now!

**Pull request merged**